.github/workflows/rust.yml

```
@@ -30,124 +30,120 @@ jobs:
       if: runner.os == 'macOS'
       run: |
         brew install openssl
-    
-    - run: cargo build --workspace --verbose --locked --release
-    - run: cargo test --workspace --verbose
 
-    # save the pyrsia_node binary for the package creation
-    - name: Save pyrsia_node binary
-      uses: actions/upload-artifact@v2
+    # Install cargo deb and strip
+    - name: Install cargo-deb cargo-strip
       if: runner.os == 'Linux'
-      with:
-        name: pyrsia_node
-        path: target/release/pyrsia_node
-        if-no-files-found: error
+      uses: actions-rs/cargo@v1.0.1
+      with: 
+        command: install
+        args: cargo-deb cargo-strip --force
 
-    # save the pyrsia binary for the package creation
-    - name: Save pyrsia binary
-      uses: actions/upload-artifact@v2
-      if: runner.os == 'Linux'
-      with:
-        name: pyrsia
-        path: target/release/pyrsia
-        if-no-files-found: error
+    # Need to add build number to version number
+    - name: Iterate Build Version Number
+      run: |
+        sed -i.bak "/^version = \"0.1.0\"/s/\"\$/+${{ github.run_number }}\"/" Cargo.toml
```

Absolutely love this 🚀 

One small grip... run number != code and this should answer "what code is being run" 

SHAs are easier to pin down and track... so I've love to tweak this

```suggestion
        sed -i.bak "/^version = \"0.1.0\"/s/\"\$/+${{ github.sha}}\"/" Cargo.toml
```

---

The build portion of the version number needs to be sequential.  That is how the apt-get installer determines if it needs to do an upgrade of a package.  The sha is not sequential.

---

Ahhh thanks for explaining that differently. it makes a lot more sense now 👍